File: linearsearch.java
import java.util.Scanner;

class Data
{
	int n;
	int elements[];
	Data(int num)
	{
		n=num;
		elements = new int[n];
	}
	void getEle()
	{
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter the elements");
		for(int i=0;i<n;i++)
		{
			elements[i] = sc.nextInt();
		}
	}
	int findEle(int ele)
	{
		for(int i=1;i<n;i++)
		{
			if(ele==elements[i])
			{
				return i;
			}
		}
		return -1;
	}

}

class LinearSearch
{
	public static void main(String args[])
	{
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter the number of numbers");
		int num = sc.nextInt();
		Data d = new Data(num);
		d.getEle();
		System.out.println("Enter the search element");
		int ele=sc.nextInt();
		int index = d.findEle(ele);
		if(index==-1)
		{
			System.out.println("element not found");
		}
		else
		{
			System.out.println("element found at index : "+index);
		}
	}
}

File: linearsearch.c
/*1. Imperative programming paradigm:
It is one of the oldest programming paradigm.
It features close relation relation to machine architecture.
It is based on Von Neumann architecture.
It works by changing the program state through assignment statements.
It performs step by step task by changing state.
The main focus is on how to achieve the goal.
The paradigm consist of several statements and after execution of all the result is stored.*/

#include<stdio.h>

int main()
{
	int i,n,ele,index;
	printf("Enter the number of numbers \n");
	scanf("%d",&n);
	int arr[n];
	printf("Enter the numbers\n");
	for(i=0;i<n;i++)
	{
		scanf("%d",&arr[i]);
	}
	printf("Enter a number to search\n");
	scanf("%d",&ele);
	index=-1;
	for(i=0;i<n;i++)
	{
		if(ele==arr[i])
		{
			index=i;
			break;
		}
	}
	if(index == -1)
	{
		printf("Element not present\n");
	}
	else
	{
		printf("Element is present at index %d\n",index);
	}

	return 0;
}